[BUG] Errors found in GraphScope standalone tryout

This is a summary of error or bug found by  GraphScope standalone tryout

- os: centos8
- python version: 3.6.8
- graphscope version: 0.8.0
- other context: test on image centos:8
- code
```bash
yum install python3-pip
pip3 install --upgrade pip
pip3 install graphscope-0.8.0-py2.py3-none-manylinux_2_17_x86_64.manylinux2014_x86_64.whl
```
- error
```bash
[root@f8a72d2f9440 wheel]# pip3 install graphscope-0.8.0-py2.py3-none-Using legacy 'setup.py install' for daemons, since package 'wheel' is not installed.
Using legacy 'setup.py install' for hdfs3, since package 'wheel' is not installed.
Using legacy 'setup.py install' for oss2, since package 'wheel' is not installed.
Using legacy 'setup.py install' for pickle5, since package 'wheel' is not installed.
Using legacy 'setup.py install' for treelib, since package 'wheel' is not installed.
Using legacy 'setup.py install' for aiobotocore, since package 'wheel' is not installed.
Using legacy 'setup.py install' for aliyun-python-sdk-core, since package 'wheel' is not installed.
Using legacy 'setup.py install' for crcmod, since package 'wheel' is not installed.
Using legacy 'setup.py install' for future, since package 'wheel' is not installed.
Using legacy 'setup.py install' for idna-ssl, since package 'wheel' is not installed.
Installing collected packages: pycparser, typing-extensions, six, multidict, idna, cffi, zipp, yarl, urllib3, python-dateutil, jmespath, idna-ssl, cryptography, chardet, attrs, async-timeout, wrapt, pytz, pyasn1, numpy, importlib-resources, importlib-metadata, future, charset-normalizer, certifi, botocore, aliyun-python-sdk-core, aioitertools, aiohttp, treelib, sortedcontainers, shared-memory38, setuptools, rsa, requests, pycryptodome, pyasn1-modules, pyarrow, pickle5, pandas, oauthlib, fsspec, etcd-distro, crcmod, cachetools, backports.zoneinfo, argcomplete, aliyun-python-sdk-kms, aiobotocore, websocket-client, vineyard, s3fs, requests-oauthlib, PyYAML, pyparsing, pyorc, protobuf, oss2, nest-asyncio, isodate, hdfs3, grpcio, google-auth, decorator, aenum, vineyard-io, tqdm, scipy, packaging, networkx, kubernetes, grpcio-tools, gremlinpython, Cython, graphscope-client, daemons, cmake, graphscope
    Running setup.py install for idna-ssl ... done
    Running setup.py install for future ... done
    Running setup.py install for aliyun-python-sdk-core ... done
    Running setup.py install for treelib ... done
  Attempting uninstall: setuptools
    Found existing installation: setuptools 39.2.0
    Uninstalling setuptools-39.2.0:
      Successfully uninstalled setuptools-39.2.0
    Running setup.py install for pickle5 ... error
    ERROR: Command errored out with exit status 1:
     command: /usr/bin/python3.6 -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-pgbnievu/pickle5_2df380b7bbaa4846abaed00e6d781be3/setup.py'"'"'; __file__='"'"'/tmp/pip-install-pgbnievu/pickle5_2df380b7bbaa4846abaed00e6d781be3/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /tmp/pip-record-3r34d4gs/install-record.txt --single-version-externally-managed --compile --install-headers /usr/local/include/python3.6m/pickle5
         cwd: /tmp/pip-install-pgbnievu/pickle5_2df380b7bbaa4846abaed00e6d781be3/
    Complete output (27 lines):
    running install
    /usr/local/lib/python3.6/site-packages/setuptools/command/install.py:37: SetuptoolsDeprecationWarning: setup.py install is deprecated. Use build and pip and other standards-based tools.
      setuptools.SetuptoolsDeprecationWarning,
    running build
    running build_py
    creating build
    creating build/lib.linux-x86_64-3.6
    creating build/lib.linux-x86_64-3.6/pickle5
    copying pickle5/pickle.py -> build/lib.linux-x86_64-3.6/pickle5
    copying pickle5/__init__.py -> build/lib.linux-x86_64-3.6/pickle5
    copying pickle5/pickletools.py -> build/lib.linux-x86_64-3.6/pickle5
    creating build/lib.linux-x86_64-3.6/pickle5/test
    copying pickle5/test/pickletester.py -> build/lib.linux-x86_64-3.6/pickle5/test
    copying pickle5/test/test_pickle.py -> build/lib.linux-x86_64-3.6/pickle5/test
    copying pickle5/test/__init__.py -> build/lib.linux-x86_64-3.6/pickle5/test
    copying pickle5/test/test_picklebuffer.py -> build/lib.linux-x86_64-3.6/pickle5/test
    running build_ext
    building 'pickle5._pickle' extension
    creating build/temp.linux-x86_64-3.6
    creating build/temp.linux-x86_64-3.6/pickle5
    gcc -pthread -Wno-unused-result -Wsign-compare -DDYNAMIC_ANNOTATIONS_ENABLED=1 -DNDEBUG -O2 -g -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fexceptions -fstack-protector-strong -grecord-gcc-switches -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -D_GNU_SOURCE -fPIC -fwrapv -O2 -g -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fexceptions -fstack-protector-strong -grecord-gcc-switches -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -D_GNU_SOURCE -fPIC -fwrapv -O2 -g -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fexceptions -fstack-protector-strong -grecord-gcc-switches -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -D_GNU_SOURCE -fPIC -fwrapv -fPIC -I/usr/include/python3.6m -c pickle5/_pickle.c -o build/temp.linux-x86_64-3.6/pickle5/_pickle.o -std=c99
    In file included from pickle5/_pickle.c:2:
    pickle5/compat.h:1:10: fatal error: Python.h: No such file or directory
     #include "Python.h"
              ^~~~~~~~~~
    compilation terminated.
    error: command 'gcc' failed with exit status 1
    ----------------------------------------
ERROR: Command errored out with exit status 1: /usr/bin/python3.6 -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-pgbnievu/pickle5_2df380b7bbaa4846abaed00e6d781be3/setup.py'"'"'; __file__='"'"'/tmp/pip-install-pgbnievu/pickle5_2df380b7bbaa4846abaed00e6d781be3/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /tmp/pip-record-3r34d4gs/install-record.txt --single-version-externally-managed --compile --install-headers /usr/local/include/python3.6m/pickle5 Check the logs for full command output.
```

> * os: centos8
> * python version: 3.6.8
> * graphscope version: 0.8.0
> * other context: test on image centos:8
> * code
> 
> ```shell
> yum install python3-pip
> pip3 install --upgrade pip
> pip3 install graphscope-0.8.0-py2.py3-none-manylinux_2_17_x86_64.manylinux2014_x86_64.whl
> ```
> 
> * error
> 
> ```shell
> [root@f8a72d2f9440 wheel]# pip3 install graphscope-0.8.0-py2.py3-none-Using legacy 'setup.py install' for daemons, since package 'wheel' is not installed.
> Using legacy 'setup.py install' for hdfs3, since package 'wheel' is not installed.
> Using legacy 'setup.py install' for oss2, since package 'wheel' is not installed.
> Using legacy 'setup.py install' for pickle5, since package 'wheel' is not installed.
> Using legacy 'setup.py install' for treelib, since package 'wheel' is not installed.
> Using legacy 'setup.py install' for aiobotocore, since package 'wheel' is not installed.
> Using legacy 'setup.py install' for aliyun-python-sdk-core, since package 'wheel' is not installed.
> Using legacy 'setup.py install' for crcmod, since package 'wheel' is not installed.
> Using legacy 'setup.py install' for future, since package 'wheel' is not installed.
> Using legacy 'setup.py install' for idna-ssl, since package 'wheel' is not installed.
> Installing collected packages: pycparser, typing-extensions, six, multidict, idna, cffi, zipp, yarl, urllib3, python-dateutil, jmespath, idna-ssl, cryptography, chardet, attrs, async-timeout, wrapt, pytz, pyasn1, numpy, importlib-resources, importlib-metadata, future, charset-normalizer, certifi, botocore, aliyun-python-sdk-core, aioitertools, aiohttp, treelib, sortedcontainers, shared-memory38, setuptools, rsa, requests, pycryptodome, pyasn1-modules, pyarrow, pickle5, pandas, oauthlib, fsspec, etcd-distro, crcmod, cachetools, backports.zoneinfo, argcomplete, aliyun-python-sdk-kms, aiobotocore, websocket-client, vineyard, s3fs, requests-oauthlib, PyYAML, pyparsing, pyorc, protobuf, oss2, nest-asyncio, isodate, hdfs3, grpcio, google-auth, decorator, aenum, vineyard-io, tqdm, scipy, packaging, networkx, kubernetes, grpcio-tools, gremlinpython, Cython, graphscope-client, daemons, cmake, graphscope
>     Running setup.py install for idna-ssl ... done
>     Running setup.py install for future ... done
>     Running setup.py install for aliyun-python-sdk-core ... done
>     Running setup.py install for treelib ... done
>   Attempting uninstall: setuptools
>     Found existing installation: setuptools 39.2.0
>     Uninstalling setuptools-39.2.0:
>       Successfully uninstalled setuptools-39.2.0
>     Running setup.py install for pickle5 ... error
>     ERROR: Command errored out with exit status 1:
>      command: /usr/bin/python3.6 -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-pgbnievu/pickle5_2df380b7bbaa4846abaed00e6d781be3/setup.py'"'"'; __file__='"'"'/tmp/pip-install-pgbnievu/pickle5_2df380b7bbaa4846abaed00e6d781be3/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /tmp/pip-record-3r34d4gs/install-record.txt --single-version-externally-managed --compile --install-headers /usr/local/include/python3.6m/pickle5
>          cwd: /tmp/pip-install-pgbnievu/pickle5_2df380b7bbaa4846abaed00e6d781be3/
>     Complete output (27 lines):
>     running install
>     /usr/local/lib/python3.6/site-packages/setuptools/command/install.py:37: SetuptoolsDeprecationWarning: setup.py install is deprecated. Use build and pip and other standards-based tools.
>       setuptools.SetuptoolsDeprecationWarning,
>     running build
>     running build_py
>     creating build
>     creating build/lib.linux-x86_64-3.6
>     creating build/lib.linux-x86_64-3.6/pickle5
>     copying pickle5/pickle.py -> build/lib.linux-x86_64-3.6/pickle5
>     copying pickle5/__init__.py -> build/lib.linux-x86_64-3.6/pickle5
>     copying pickle5/pickletools.py -> build/lib.linux-x86_64-3.6/pickle5
>     creating build/lib.linux-x86_64-3.6/pickle5/test
>     copying pickle5/test/pickletester.py -> build/lib.linux-x86_64-3.6/pickle5/test
>     copying pickle5/test/test_pickle.py -> build/lib.linux-x86_64-3.6/pickle5/test
>     copying pickle5/test/__init__.py -> build/lib.linux-x86_64-3.6/pickle5/test
>     copying pickle5/test/test_picklebuffer.py -> build/lib.linux-x86_64-3.6/pickle5/test
>     running build_ext
>     building 'pickle5._pickle' extension
>     creating build/temp.linux-x86_64-3.6
>     creating build/temp.linux-x86_64-3.6/pickle5
>     gcc -pthread -Wno-unused-result -Wsign-compare -DDYNAMIC_ANNOTATIONS_ENABLED=1 -DNDEBUG -O2 -g -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fexceptions -fstack-protector-strong -grecord-gcc-switches -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -D_GNU_SOURCE -fPIC -fwrapv -O2 -g -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fexceptions -fstack-protector-strong -grecord-gcc-switches -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -D_GNU_SOURCE -fPIC -fwrapv -O2 -g -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fexceptions -fstack-protector-strong -grecord-gcc-switches -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -D_GNU_SOURCE -fPIC -fwrapv -fPIC -I/usr/include/python3.6m -c pickle5/_pickle.c -o build/temp.linux-x86_64-3.6/pickle5/_pickle.o -std=c99
>     In file included from pickle5/_pickle.c:2:
>     pickle5/compat.h:1:10: fatal error: Python.h: No such file or directory
>      #include "Python.h"
>               ^~~~~~~~~~
>     compilation terminated.
>     error: command 'gcc' failed with exit status 1
>     ----------------------------------------
> ERROR: Command errored out with exit status 1: /usr/bin/python3.6 -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-pgbnievu/pickle5_2df380b7bbaa4846abaed00e6d781be3/setup.py'"'"'; __file__='"'"'/tmp/pip-install-pgbnievu/pickle5_2df380b7bbaa4846abaed00e6d781be3/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /tmp/pip-record-3r34d4gs/install-record.txt --single-version-externally-managed --compile --install-headers /usr/local/include/python3.6m/pickle5 Check the logs for full command output.
> ```

This is because vineyard requires pickle5 for `python <= 3.7`, which further requires `python-devel` package been installed to compile the target.

The issue has been tracked at https://github.com/pitrou/pickle5-backport/issues/24

--- a/.github/workflows/build-graphscope-wheels-linux.yml
+++ b/.github/workflows/build-graphscope-wheels-linux.yml
@@ -104,7 +104,7 @@ jobs:
         popd
 
         # run test
-        pip3 install pytest tensorflow --user;
+        pip3 install pytest tensorflow==2.6.0 --user;
         sudo apt update -y && sudo apt install openjdk-11-jdk -y
         cp ${GITHUB_WORKSPACE}/python/tests/local/test_standalone.py /tmp/test_standalone.py
         python3 -m pytest -s -v /tmp/test_standalone.py

--- a/.github/workflows/build-graphscope-wheels-linux.yml
+++ b/.github/workflows/build-graphscope-wheels-linux.yml
@@ -104,7 +104,7 @@ jobs:
         popd
 
         # run test
-        pip3 install pytest tensorflow<=2.5.2 --user;
+        pip3 install pytest "tensorflow<=2.5.2" --user;
         sudo apt update -y && sudo apt install openjdk-11-jdk -y
         cp ${GITHUB_WORKSPACE}/python/tests/local/test_standalone.py /tmp/test_standalone.py
         python3 -m pytest -s -v /tmp/test_standalone.py

--- a/.github/workflows/build-graphscope-wheels-linux.yml
+++ b/.github/workflows/build-graphscope-wheels-linux.yml
@@ -104,7 +104,7 @@ jobs:
         popd
 
         # run test
-        pip3 install pytest tensorflow==2.6.0 --user;
+        pip3 install pytest tensorflow<=2.5.2 --user;
         sudo apt update -y && sudo apt install openjdk-11-jdk -y
         cp ${GITHUB_WORKSPACE}/python/tests/local/test_standalone.py /tmp/test_standalone.py
         python3 -m pytest -s -v /tmp/test_standalone.py

--- a/.github/workflows/build-graphscope-wheels-macos.yml
+++ b/.github/workflows/build-graphscope-wheels-macos.yml
@@ -110,7 +110,7 @@ jobs:
 
         # run test
         # waiting for graphlearn supported on macos
-        # pip3 install pytest tensorflow<=2.5.2 --user;
+        # pip3 install pytest "tensorflow<=2.5.2" --user;
         pip3 install pytest --user
         export PATH=/Users/runner/Library/Python/3.9/bin:${PATH}
         cp ${GITHUB_WORKSPACE}/python/tests/local/test_standalone.py /tmp/test_standalone.py

--- a/.github/workflows/build-graphscope-wheels-macos.yml
+++ b/.github/workflows/build-graphscope-wheels-macos.yml
@@ -110,7 +110,7 @@ jobs:
 
         # run test
         # waiting for graphlearn supported on macos
-        # pip3 install pytest tensorflow --user;
+        # pip3 install pytest tensorflow==2.6.0 --user;
         pip3 install pytest --user
         export PATH=/Users/runner/Library/Python/3.9/bin:${PATH}
         cp ${GITHUB_WORKSPACE}/python/tests/local/test_standalone.py /tmp/test_standalone.py

--- a/.github/workflows/build-graphscope-wheels-macos.yml
+++ b/.github/workflows/build-graphscope-wheels-macos.yml
@@ -110,7 +110,7 @@ jobs:
 
         # run test
         # waiting for graphlearn supported on macos
-        # pip3 install pytest tensorflow==2.6.0 --user;
+        # pip3 install pytest tensorflow<=2.5.2 --user;
         pip3 install pytest --user
         export PATH=/Users/runner/Library/Python/3.9/bin:${PATH}
         cp ${GITHUB_WORKSPACE}/python/tests/local/test_standalone.py /tmp/test_standalone.py

--- a/.github/workflows/local.yml
+++ b/.github/workflows/local.yml
@@ -68,7 +68,7 @@ jobs:
         # install tensorflow for graphlearn if os is ubuntu20.04
         if [ "${OS}" == "ubuntu-20.04" ];
         then
-            pip3 install pytest tensorflow<=2.5.2 --user;
+            pip3 install pytest "tensorflow<=2.5.2" --user;
         fi
 
         export GS_TEST_DIR=${GITHUB_WORKSPACE}/gstest

--- a/.github/workflows/local.yml
+++ b/.github/workflows/local.yml
@@ -68,7 +68,7 @@ jobs:
         # install tensorflow for graphlearn if os is ubuntu20.04
         if [ "${OS}" == "ubuntu-20.04" ];
         then
-            pip3 install pytest tensorflow --user;
+            pip3 install pytest tensorflow==2.6.0 --user;
         fi
 
         export GS_TEST_DIR=${GITHUB_WORKSPACE}/gstest

--- a/.github/workflows/local.yml
+++ b/.github/workflows/local.yml
@@ -68,7 +68,7 @@ jobs:
         # install tensorflow for graphlearn if os is ubuntu20.04
         if [ "${OS}" == "ubuntu-20.04" ];
         then
-            pip3 install pytest tensorflow==2.6.0 --user;
+            pip3 install pytest tensorflow<=2.5.2 --user;
         fi
 
         export GS_TEST_DIR=${GITHUB_WORKSPACE}/gstest

--- a/Makefile
+++ b/Makefile
@@ -1,6 +1,5 @@
 MKFILE_PATH 			:= $(abspath $(lastword $(MAKEFILE_LIST)))
 WORKING_DIR 			:= $(dir $(MKFILE_PATH))
-NUM_PROC                := $( $(command -v nproc &> /dev/null) && echo $(nproc) || echo $(sysctl -n hw.physicalcpu) )
 
 VERSION                     ?= 0.1.0
 INSTALL_PREFIX              ?= /opt/graphscope
@@ -62,7 +61,7 @@ gae:
 	mkdir -p $(WORKING_DIR)/analytical_engine/build
 	cd $(WORKING_DIR)/analytical_engine/build && \
 	cmake -DCMAKE_INSTALL_PREFIX=$(INSTALL_PREFIX) -DNETWORKX=$(NETWORKX) -DBUILD_TESTS=${BUILD_TEST} .. && \
-	make -j$(NUM_PROC) && \
+	make -j`nproc` && \
 	sudo make install
 ifneq ($(INSTALL_PREFIX), /usr/local)
 	sudo rm -fr /usr/local/include/graphscope && \
@@ -110,7 +109,7 @@ ifeq ($(WITH_LEARNING_ENGINE), ON)
 	git submodule update --init third_party/pybind11 && \
 	mkdir -p cmake-build && cd cmake-build && \
 	cmake -DCMAKE_INSTALL_PREFIX=$(INSTALL_PREFIX) -DWITH_VINEYARD=ON -DTESTING=${BUILD_TEST} .. && \
-	make -j$(NUM_PROC) && \
+	make -j`nproc` && \
 	sudo make install
 ifneq ($(INSTALL_PREFIX), /usr/local)
 	sudo ln -sf ${INSTALL_PREFIX}/lib/*so* /usr/local/lib

--- a/coordinator/gscoordinator/utils.py
+++ b/coordinator/gscoordinator/utils.py
@@ -1337,6 +1337,9 @@ class ResolveMPICmdPrefix(object):
 
         if num_workers == 1 and (hosts == "localhost" or hosts == "127.0.0.1"):
             # run without mpi on localhost if workers num is 1
+            if shutil.which("ssh") is None:
+                # also need a fake ssh agent
+                env[self._OPENMPI_RSH_AGENT] = sys.executable
             return cmd, env
 
         if self.openmpi():

--- a/coordinator/requirements.txt
+++ b/coordinator/requirements.txt
@@ -1,7 +1,7 @@
 cmake>=3.21.2
 daemons
 etcd-distro>=3.5.1
-graphscope-client
+graphscope-client>=0.8.0
 grpcio
 kubernetes
 protobuf

--- a/coordinator/requirements.txt
+++ b/coordinator/requirements.txt
@@ -1,7 +1,7 @@
 cmake>=3.21.2
 daemons
 etcd-distro>=3.5.1
-graphscope-client>=0.8.0
+graphscope-client==0.8.0
 grpcio
 kubernetes
 protobuf

--- a/k8s/Makefile
+++ b/k8s/Makefile
@@ -154,6 +154,8 @@ graphscope-manylinux2014-py3:
 			export WITH_EXTRA_DATA=ON && \
 			cd /work/coordinator && \
 			rm -fr build dist/*.whl && \
+			strip /opt/graphscope/bin/executor && \
+			strip /opt/graphscope/bin/gaia_executor && \
 			python3 setup.py bdist_wheel && \
 			cd dist && \
 			auditwheel repair ./*.whl --plat=manylinux2014_x86_64'

--- a/k8s/Makefile
+++ b/k8s/Makefile
@@ -154,8 +154,8 @@ graphscope-manylinux2014-py3:
 			export WITH_EXTRA_DATA=ON && \
 			cd /work/coordinator && \
 			rm -fr build dist/*.whl && \
-			strip /opt/graphscope/bin/executor && \
-			strip /opt/graphscope/bin/gaia_executor && \
+			sudo strip /opt/graphscope/bin/executor && \
+			sudo strip /opt/graphscope/bin/gaia_executor && \
 			python3 setup.py bdist_wheel && \
 			cd dist && \
 			auditwheel repair ./*.whl --plat=manylinux2014_x86_64'

--- a/python/graphscope/framework/utils.py
+++ b/python/graphscope/framework/utils.py
@@ -47,7 +47,13 @@ def is_free_port(port, host="localhost", timeout=0.2):
         True if port is available, False otherwise.
     """
     if host == "localhost" or host == "127.0.0.1":
-        return int(port) not in [conn.laddr.port for conn in psutil.net_connections()]
+        try:
+            return int(port) not in [
+                conn.laddr.port for conn in psutil.net_connections()
+            ]
+        except psutil.AccessDenied:
+            # back to the socket.connect
+            pass
 
     sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
     try:

--- a/python/graphscope/framework/utils.py
+++ b/python/graphscope/framework/utils.py
@@ -24,6 +24,7 @@ import string
 
 import numpy as np
 import pandas as pd
+import psutil
 from google.protobuf.any_pb2 import Any
 
 from graphscope.client.archive import OutArchive
@@ -45,6 +46,9 @@ def is_free_port(port, host="localhost", timeout=0.2):
     Returns:
         True if port is available, False otherwise.
     """
+    if host == "localhost" or host == "127.0.0.1":
+        return int(port) not in [conn.laddr.port for conn in psutil.net_connections()]
+
     sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
     try:
         sock.settimeout(timeout)

--- a/python/requirements.txt
+++ b/python/requirements.txt
@@ -9,6 +9,7 @@ numpy
 packaging
 pandas
 protobuf>=3.12.0
+psutil
 PyYAML
 scipy
 tqdm

--- a/scripts/deploy_local.sh
+++ b/scripts/deploy_local.sh
@@ -558,7 +558,7 @@ install_dependencies() {
       check_and_remove_dir "/tmp/openmpi-4.0.5"
       tar zxvf /tmp/openmpi-4.0.5.tar.gz -C /tmp
       pushd /tmp/openmpi-4.0.5 && ./configure --enable-mpi-cxx
-      make -j`nproc`
+      make -j$(nproc)
       sudo make install
       popd
       rm -fr /tmp/openmpi-4.0.5 /tmp/openmpi-4.0.5.tar.gz
@@ -615,7 +615,7 @@ install_dependencies() {
     tar zxvf /tmp/protobuf-all-3.13.0.tar.gz -C /tmp/
     pushd /tmp/protobuf-3.13.0
     ./configure --enable-shared --disable-static
-    make -j`nproc`
+    make -j$(nproc)
     sudo make install && ldconfig
     popd
     rm -fr /tmp/protobuf-all-3.13.0.tar.gz /tmp/protobuf-3.13.0
@@ -642,7 +642,7 @@ install_dependencies() {
         -DgRPC_PROTOBUF_PROVIDER=package \
         -DgRPC_ZLIB_PROVIDER=package \
         -DgRPC_SSL_PROVIDER=package
-    make -j`nproc`
+    make -j$(nproc)
     sudo make install
     popd
     rm -fr /tmp/grpc
@@ -710,7 +710,7 @@ install_dependencies() {
       pushd /tmp/fmt-7.0.3
       mkdir -p build && cd build
       cmake .. -DBUILD_SHARED_LIBS=ON
-      make -j`nproc`
+      make -j$(nproc)
       sudo make install
       popd
       rm -fr /tmp/7.0.3.tar.gz /tmp/fmt-7.0.3
@@ -722,7 +722,7 @@ install_dependencies() {
     pushd /tmp/folly-2020.10.19.00
     mkdir -p _build && cd _build
     cmake -DBUILD_SHARED_LIBS=ON -DCMAKE_POSITION_INDEPENDENT_CODE=ON ..
-    make -j`nproc`
+    make -j$(nproc)
     sudo make install
     popd
     rm -fr /tmp/v2020.10.19.00.tar.gz /tmp/folly-2020.10.19.00
@@ -759,7 +759,7 @@ install_libgrape-lite() {
   pushd /tmp/libgrape-lite
   mkdir -p build && cd build
   cmake ..
-  make -j`nproc`
+  make -j$(nproc)
   sudo make install
   popd
   rm -fr /tmp/libgrape-lite
@@ -798,8 +798,8 @@ install_vineyard() {
     cmake .. -DBUILD_SHARED_LIBS=ON \
              -DBUILD_VINEYARD_TESTS=OFF
   fi
-  make -j`nproc`
-  make vineyard_client_python -j`nproc`
+  make -j$(nproc)
+  make vineyard_client_python -j$(nproc)
   sudo make install
   popd
 

--- a/scripts/deploy_local.sh
+++ b/scripts/deploy_local.sh
@@ -18,7 +18,6 @@ readonly V6D_VERSION="0.3.4"  # vineyard version
 readonly V6D_BRANCH="v0.3.4" # vineyard branch
 
 readonly SOURCE_DIR="$( cd "$(dirname $0)/.." >/dev/null 2>&1 ; pwd -P )"
-readonly NUM_PROC=$( $(command -v nproc &> /dev/null) && echo $(nproc) || echo $(sysctl -n hw.physicalcpu) )
 readonly OUTPUT_ENV_FILE="${HOME}/.graphscope_env"
 IS_IN_WSL=false && [[ ! -z "${IS_WSL}" || ! -z "${WSL_DISTRO_NAME}" ]] && IS_IN_WSL=true
 readonly IS_IN_WSL
@@ -255,6 +254,7 @@ init_basic_packages() {
     )
   else
     BASIC_PACKGES_TO_INSTALL=(
+      coreutils
       double-conversion
       protobuf
       glog
@@ -558,7 +558,7 @@ install_dependencies() {
       check_and_remove_dir "/tmp/openmpi-4.0.5"
       tar zxvf /tmp/openmpi-4.0.5.tar.gz -C /tmp
       pushd /tmp/openmpi-4.0.5 && ./configure --enable-mpi-cxx
-      make -j${NUM_PROC}
+      make -j`nproc`
       sudo make install
       popd
       rm -fr /tmp/openmpi-4.0.5 /tmp/openmpi-4.0.5.tar.gz
@@ -615,7 +615,7 @@ install_dependencies() {
     tar zxvf /tmp/protobuf-all-3.13.0.tar.gz -C /tmp/
     pushd /tmp/protobuf-3.13.0
     ./configure --enable-shared --disable-static
-    make -j${NUM_PROC}
+    make -j`nproc`
     sudo make install && ldconfig
     popd
     rm -fr /tmp/protobuf-all-3.13.0.tar.gz /tmp/protobuf-3.13.0
@@ -642,7 +642,7 @@ install_dependencies() {
         -DgRPC_PROTOBUF_PROVIDER=package \
         -DgRPC_ZLIB_PROVIDER=package \
         -DgRPC_SSL_PROVIDER=package
-    make -j${NUM_PROC}
+    make -j`nproc`
     sudo make install
     popd
     rm -fr /tmp/grpc
@@ -710,7 +710,7 @@ install_dependencies() {
       pushd /tmp/fmt-7.0.3
       mkdir -p build && cd build
       cmake .. -DBUILD_SHARED_LIBS=ON
-      make -j${NUM_PROC}
+      make -j`nproc`
       sudo make install
       popd
       rm -fr /tmp/7.0.3.tar.gz /tmp/fmt-7.0.3
@@ -722,7 +722,7 @@ install_dependencies() {
     pushd /tmp/folly-2020.10.19.00
     mkdir -p _build && cd _build
     cmake -DBUILD_SHARED_LIBS=ON -DCMAKE_POSITION_INDEPENDENT_CODE=ON ..
-    make -j${NUM_PROC}
+    make -j`nproc`
     sudo make install
     popd
     rm -fr /tmp/v2020.10.19.00.tar.gz /tmp/folly-2020.10.19.00
@@ -740,8 +740,6 @@ install_dependencies() {
 
 ##########################
 # Install libgrape-lite.
-# Globals:
-#   NUM_PROC
 # Arguments:
 #   None
 # Outputs:
@@ -761,7 +759,7 @@ install_libgrape-lite() {
   pushd /tmp/libgrape-lite
   mkdir -p build && cd build
   cmake ..
-  make -j${NUM_PROC}
+  make -j`nproc`
   sudo make install
   popd
   rm -fr /tmp/libgrape-lite
@@ -771,7 +769,6 @@ install_libgrape-lite() {
 # Install vineyard.
 # Globals:
 #   PLATFORM
-#   NUM_PROC
 # Arguments:
 #   None
 # Outputs:
@@ -801,8 +798,8 @@ install_vineyard() {
     cmake .. -DBUILD_SHARED_LIBS=ON \
              -DBUILD_VINEYARD_TESTS=OFF
   fi
-  make -j${NUM_PROC}
-  make vineyard_client_python -j${NUM_PROC}
+  make -j`nproc`
+  make vineyard_client_python -j`nproc`
   sudo make install
   popd
 
@@ -826,7 +823,6 @@ install_vineyard() {
 # Globals:
 #   SOURCE_DIR
 #   INSTALL_PREFIX
-#   NUM_PROC
 # Arguments:
 #   None
 # Outputs:

